### PR TITLE
tests(utils): add unit tests for murmurhash3 implementation

### LINKED ISSUE #1528: tests(utils): add unit tests for murmurhash3 implementation
### Description
Add comprehensive unit tests for MurmurHash3 implementation
The `pkg/utils/hash/murmur3.go` file currently lacks unit tests. We need comprehensive test coverage to ensure the hash function's correctness and reliability.

### PR DESCRIPTION
**Pull request by shivansh-gohem**
Description:
Adds comprehensive table-driven unit tests for the MurmurHash3 implementation in pkg/utils/hash.

Test Coverage includes:

(i)Empty inputs

(ii)Inputs smaller than the chunk size

(iii)Inputs matching exact chunk boundaries

(iv)Large inputs with special characters

Fixes #1528

### CODE REVIEW

File: pkg/utils/hash/murmur3_test.go [hunk 1]
```diff
@@ -120,30 +120,33 @@ func TestFmix64(t *testing.T) {
...
+		{
+			name:     "test value 1",
+			input:    0x0123456789abcdef,
+			expected: 0x87cbfbfe89022cea,
 		},
 		{
-			name:     "test value",
-			input:    0x123456789abcdef0,
-			expected: 0, // This would need to be calculated/verified - testing for determinism instead
+			name:     "test value 2",
```
Comment (LiZhenCheng9527) [inline pkg/utils/hash/murmur3_test.go]: Why test this use case? I don't see much value in this test.
This case does not extend the scope of the tests covered.

**Comment by shivansh-gohem** [comment]
@LiZhenCheng9527 I've removed those extra test cases as suggested and updated the PR. Could you please take another look?

Also, as a heads-up, the CI build keeps failing on Test_linkXdp in the pkg/controller/manage package, which seems like an unrelated flaky test in the infrastructure. The local pkg/utils/hash tests are passing cleanly!

**Comment by shivansh-gohem** [comment]
@LiZhenCheng9527 i think the flaky behaviour by build is not curable from my side!! i have resolved all the required changes!

**Comment by LiZhenCheng9527** [comment]
> @LiZhenCheng9527 i think the flaky behaviour by build is not curable from my side!! i have resolved all the required changes!

Then I don't understand what the point of this PR is. 
In my opinion, "max value" is more appropriate than "all one."

**Comment by shivansh-gohem** [comment]
Hi @LiZhenCheng9527, my apologies for the back-and-forth. I have reverted the name back to 'max value' in the latest commit as you requested! (I am ignoring the AI bot suggestions here to focus on your feedback)

**Comment by shivansh-gohem** [comment]
@LiZhenCheng9527 To clarify the overall goal of this PR: it aims to improve the unit test coverage for pkg/utils/hash by covering edge cases like empty inputs, specific chunk boundaries, and large inputs, ensuring the MurmurHash3 implementation remains stable against regressions
